src/app/authentication/login.ts

```diff
@@ -362,12 +364,15 @@ export const loginWithDevice = async ({
       return ipFetcherInfo
     }
 
-    const validatedIPMetadata = IPMetadataValidator(
+    const authorizedIPMetadata = IPMetadataAuthorizer(
       ipMetadataValidationSettings,
-    ).validate(ipFetcherInfo)
+    ).authorize(ipFetcherInfo)
 
-    if (validatedIPMetadata instanceof Error) {
-      return new InvalidIPForOnboardingError(validatedIPMetadata.name)
+    if (authorizedIPMetadata instanceof Error) {
+      if (authorizedIPMetadata instanceof MissingIPMetadataError)
+        return new InvalidIpMetadataError(authorizedIPMetadata.name)
```

Comment:
Why are we only passing the name on? Aren't we potentially silently dropping interesting error context?

Reply:
yeah, this was the previous behavior. just changed it.